src/hooks/room/useRoomCall.ts

```diff
@@ -279,7 +279,7 @@ export const useRoomCall = (
         url.pathname = "/room/";
         // Set params for the sharable url
         url.searchParams.set("roomId", room.roomId);
-        url.searchParams.set("perParticipantE2EE", "true");
+        if (room.hasEncryptionStateEvent()) url.searchParams.set("perParticipantE2EE", "true");
```

Comment:
This is different from the logic used for the widget URL: https://github.com/matrix-org/matrix-react-sdk/blob/develop/src/models/Call.ts#L784 - the other one seems more correct. Shouldn't they just share a function to decide whether the call is encrypted, and/or some common code to generate the common parts of the URL?

Reply:
The other one says it's deprecated. Maybe updating both makes sense though.

Reply:
This is an sync fn however. that would trickle down quiet far. I would need to make `Call.get` async (because that needs the widget, which needs the widget url parameters that contain on the encryption flag. Is this what we want?
Its also super important, that both clients have the same understanding of the current encryption state of the room. Could CryptoApi remembering that state one one client but another client does not have this information bring this out of sync?

Reply:
This seems to break room navigation. It always takes jumping back and forth before the correct timeline is shown (in video rooms it first shows the timeline and then the video room. In normal rooms it first shows a join view and then the timeline)
I am agian thinking using the state event might be the better solution?

Reply:
I think i found a fix for it: https://github.com/matrix-org/matrix-react-sdk/pull/12345/commits/6c03bb438259c938f84e58696b58a7df333facc8

Reply:
Eugh, sorry - I didn't notice this was async, and yes, having Call.get be async is not ideal. Chatting to the crypto team suggests that actually the presence of the state event might be fine after all, so I think whichever you think do whatever you think is best.

Reply:
I updated back to not be async and use `hasEncryptionStateEvent`

Reply:
Thanks, sorry for the run-around

Reply:
no worries. I think it was worth investigating if it can be done using the async fn.